textworld/generator/game.py

```diff
@@ -137,42 +222,273 @@ def set_conditions(self, conditions: Iterable[Proposition]) -> Action:
 
         Args:
-            data: Serialized data with the needed information to build a
-                  `Event` object.
+            data: Serialized data with the needed information to build a `EventCondition` object.
         """
-        actions = [Action.deserialize(d) for d in data["actions"]]
         condition = Action.deserialize(data["condition"])
-        event = cls(actions, condition.preconditions, data["commands"])
-        return event
+        return cls(conditions=condition.preconditions)
 
     def serialize(self) -> Mapping:
         """ Serialize this event.
 
         Results:
-            `Event`'s data serialized to be JSON compatible.
+            `EventCondition`'s data serialized to be JSON compatible.
         """
-        data = {}
-        data["commands"] = self.commands
-        data["actions"] = [action.serialize() for action in self.actions]
+        data = super().serialize()
         data["condition"] = self.condition.serialize()
         return data
 
-    def copy(self) -> "Event":
-        """ Copy this event. """
+    # def copy(self) -> "EventCondition":
+    #     """ Copy this event. """
+    #     return self.deserialize(self.serialize())
+
+
+class Event:  # For backward compatibility.
+    """
+    Event happening in TextWorld.
+
+    An event gets triggered when its set of conditions become all statisfied.
+
+    .. warning:: Deprecated in favor of :py:class:`textworld.generator.EventCondition <textworld.generator.game.EventCondition>`.
+    """
+
+    def __new__(cls, actions: Iterable[Action] = (),
+                     conditions: Iterable[Proposition] = (),
+                     commands: Iterable[str] = ()):
+        return EventCondition(actions=actions, conditions=conditions, commands=commands)
+
+
+class EventAction(AbstractEvent):
+
+    def __init__(self, action: Action,
+                 actions: Iterable[Action] = (),
+                 commands: Iterable[str] = (),
+                 **kwargs) -> None:
+        """
+        Args:
+            action: The action to be performed to trigger this event.
+            actions: The actions to be performed to trigger this event.
+            commands: Human readable version of the actions.
+
+        Notes:
+            TODO: EventAction are temporal.
+        """
+        super(EventAction, self).__init__(actions, commands, **kwargs)
+        self.action = action
+
+    def is_triggering(self, state: Optional[State] = None, action: Optional[Action] = None, callback: Optional[callable] = None) -> bool:
+        """ Check if this event would be triggered for a given action. """
+        if action is None:
+            return False
+
+        mapping = self.action.match(action)
+        if mapping is None:
+            return False
+
+        is_triggering = all(ph.name == mapping[ph].name for ph in self.action.placeholders if ph.name != ph.type)
+        if callback and is_triggering:
+            callback(self)
+
+        return is_triggering
```

Comment:
@HakiRose this is how I checked the rule (of EventAction) matches the just performed action.